sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -1045,8 +1057,25 @@ internal void Simulate(float deltaTime)
 
             SimulationProfiler = Profiler.Begin(PhysicsProfilingKeys.SimulationProfilingKey);
 
-            if (discreteDynamicsWorld != null) discreteDynamicsWorld.StepSimulation(deltaTime, MaxSubSteps, FixedTimeStep);
-            else collisionWorld.PerformDiscreteCollisionDetection();
+            if (discreteDynamicsWorld != null)
+            {
+                carriedDelta += deltaTime;
+                for (; carriedDelta > FixedTimeStep; )
+                {
+                    var ts = Stopwatch.GetTimestamp();
+                    discreteDynamicsWorld.StepSimulation(FixedTimeStep, 0, FixedTimeStep);
+                    carriedDelta -= FixedTimeStep;
+                    if (carriedDelta > FixedTimeStep && (Stopwatch.GetTimestamp() - ts) / (double)Stopwatch.Frequency > MaxTickDuration)
```

Comment:
Since Stopwatch.Frequency is constant, is it possible to replace this division with multiplication of static field = 1.0 /  Stopwatch.Frequency computed once? Or is it not really worth it?

Reply:
Yep, done